src/pocketdb/models/dto/AdPost.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2018-2022 The Pocketnet developers
+// Distributed under the Apache 2.0 software license, see the accompanying
+// https://www.apache.org/licenses/LICENSE-2.0
+
+#include <primitives/transaction.h>
+#include "pocketdb/models/dto/AdPost.h"
+
+namespace PocketTx
+{
+    AdPost::AdPost() : Transaction()
+    {
+        SetType(TxType::ACTION_AD_POST);
+    }
+
+    AdPost::AdPost(const CTransactionRef& tx) : Transaction(tx)
+    {
+        SetType(TxType::ACTION_AD_POST);
+    }
+
+    shared_ptr <UniValue> AdPost::Serialize() const
```

Comment:
I recently implemented an intermediate class for "social network" transactions that implements a single serialization mechanism for RPC and node synchronization. Also consistent with DB. I suggest using this approach in new transactions, and rewriting the old ones as far as possible

https://github.com/pocketnetteam/pocketnet.core/blob/d756ee0a72673ef8524a6708571becc4e36c53d3/src/pocketdb/models/dto/moderation/Flag.h#L12

Reply:
I'll incorporate that once the change is merged over to the 0.21 branch